sources/models/src/modeled_types/kubernetes.rs

```diff
@@ -768,3 +768,102 @@ mod test_kubernetes_cloud_provider {
         }
     }
 }
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+/// CpuManagerPolicy represents a string that contains a valid cpu management policy. Default: none
+/// https://kubernetes.io/docs/tasks/administer-cluster/cpu-management-policies/
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct CpuManagerPolicy {
+    inner: String,
+}
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Deserialize)]
+#[serde(rename_all = "lowercase")]
+enum ValidCpuManagerPolicy {
+    Static,
+    None,
+}
+
+impl TryFrom<&str> for CpuManagerPolicy {
+    type Error = error::Error;
+
+    fn try_from(input: &str) -> Result<Self, Self::Error> {
+        serde_plain::from_str::<ValidCpuManagerPolicy>(&input).context(
+            error::InvalidCpuManagerPolicy { input })?;
+        Ok(CpuManagerPolicy {
+            inner: input.to_string(),
+        })
+    }
+}
+string_impls_for!(CpuManagerPolicy, "CpuManagerPolicy");
+
+#[cfg(test)]
+mod test_cpu_manager_policy {
+    use super::CpuManagerPolicy;
+    use std::convert::TryFrom;
+
+    #[test]
+    fn good_cpu_manager_policy() {
+        for ok in &["static", "none"] {
+            CpuManagerPolicy::try_from(*ok).unwrap();
+        }
+    }
+
+    #[test]
+    fn bad_cpu_manager_policy() {
+        for err in &["", "bad", "100", &"a".repeat(64)] {
+            CpuManagerPolicy::try_from(*err).unwrap_err();
+        }
+    }
+}
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+/// KubernetesDurationValue represents a string that contains a valid kubernetes duration value.
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct KubernetesDurationValue {
+    inner: String,
+}
+
+lazy_static! {
+    pub(crate) static ref CPU_MANAGER_RECONCILE_PERIOD: Regex = Regex::new(
+        r"^(([0-9.]+)(h))?(([0-9.]+)(m))?(([0-9.]+)(s))?$"
```

Comment:
Do we need all the inner capture groups? We should also support `ms`. Not sure if we should allow cases like "....3h" either. Does kubelet have a regex we could use instead?

```suggestion
        r"^([0-9.]+h)?([0-9.]+m)?([0-9.]+s)?([0-9.]+ms)?$"
```

Reply:
I don't think kubelet have a regex that we can use since I did not find anything on kubelet source code. I dot not think we should allow cases like `10s10h` in bottlerocket. Although `10s10h` is accepted in kubelet and it will be converted to `10h10s`, `10s10h` is not a meaningful format for me and we should keep a meaningful input standard .